visualization/include/pcl/visualization/impl/point_cloud_color_handlers.hpp

```diff
@@ -564,6 +485,158 @@ PointCloudColorHandlerLabelField<PointT>::getColor () const
+
+  // if any missing, error
+  if (x_field_idx == -1 || y_field_idx == -1 || z_field_idx == -1) {
+    throw std::out_of_range("isXYZFiniteAt(): input cloud missing at least one of x, y, z fields");
+  }
+  // get x,y,z field values
+  const auto position_x = index * cloud.point_step + cloud.fields[x_field_idx].offset;
+  const auto position_y = index * cloud.point_step + cloud.fields[y_field_idx].offset;
+  const auto position_z = index * cloud.point_step + cloud.fields[z_field_idx].offset;
+  if (cloud.data.size () >= (position_x + sizeof(float)) &&
+      cloud.data.size () >= (position_y + sizeof(float)) &&
+      cloud.data.size () >= (position_z + sizeof(float))) {
+    const float x = *reinterpret_cast<const float*>(&cloud.data[position_x]);
+    const float y = *reinterpret_cast<const float*>(&cloud.data[position_y]);
+    const float z = *reinterpret_cast<const float*>(&cloud.data[position_z]);
+    return isXYZFinite(PointXYZ(x, y, z));
+  } else {
+    // the last of the three is out of range
+    throw std::out_of_range("isXYZFiniteAt(): requested for index larger than number of points");
+  }
+}
+
+template <typename PointT>
+PointCloudColorHandlerGenericField<PointT>::PointCloudColorHandlerGenericField(
+    const PointCloudConstPtr& cloud, const std::string& field_name)
+: PointCloudColorHandler<PointT>(cloud), field_name_(field_name)
+{
+  this->setInputCloud(cloud);
+}
+
+template <typename PointT>
+PointCloudColorHandlerGenericField<PointT>::PointCloudColorHandlerGenericField(const std::string& field_name)
+  : PointCloudColorHandler<PointT>(), field_name_(field_name) {}
+
+template <typename PointT> void PointCloudColorHandlerGenericField<PointT>::setInputCloud(
+    const PointCloudConstPtr& cloud)
+{
+  PointCloudColorHandler<PointT>::setInputCloud(cloud);
+  this->fields_ = getFields(*cloud);
+  this->field_idx_ = getFieldIndex(field_name_, this->fields_);
+  this->capable_ = this->field_idx_ != -1;
+}
+
+template <typename PointT> std::string PointCloudColorHandlerGenericField<PointT>::getFieldName() const
+{
+  return field_name_;
+}
+
+template <typename PointT> vtkSmartPointer<vtkDataArray> PointCloudColorHandlerGenericField<PointT>::getColor() const
+{
+  if (!this->capable_ || !this->cloud_) {
+    return nullptr;
+  }
+
+  auto scalars = vtkSmartPointer<vtkFloatArray>::New();
+  scalars->SetNumberOfComponents(1);
+
+  const vtkIdType nr_points = this->cloud_->width * this->cloud_->height;
+  scalars->SetNumberOfTuples(nr_points);
+
+  // per-point color as a float value. vtk turns that into rgb somehow
+  float* colors = new float[nr_points];
+
+  // Find X channel
+  int x_channel_idx = -1;
+  for (std::size_t channel_idx = 0; channel_idx < this->fields_.size(); ++channel_idx) {
+    if (this->fields_[channel_idx].name == "x") {
+      x_channel_idx = static_cast<int>(channel_idx);
+    }
+  }
+
+  int point_offset = this->fields_[this->field_idx_].offset;
```

Comment:
A rather minor thing, but could make sense to use `std::size_t` here to also make it work with larger clouds (because it holds the offset in bytes, for every point).